### PR TITLE
BOT-314: empty string is also considered as utterance which is wrong

### PR DESCRIPTION
**Pull request by stefan-sev**
__Pull Request addresses Issue/Bug:__

What issue/bug is addressed by this pull request ?

__Implemented behaviour:__

What did you actually change ? 

__Build successful?__

Is build still possible ? ("npm run build")

__Unit Tests changed/added:__

Have unit tests been changed or added to reflect new functionality ?

### CODE REVIEW

File: src/scripting/ScriptingProvider.js [hunk 1]
```diff
@@ -257,14 +257,25 @@ module.exports = class ScriptingProvider {
...
     if (fileUtterances) {
-      fileUtterances.forEach((fileUtt) => {
-        fileUtt.sourceTag = { filename }
-      })
+      this.fileUtterances = this._tagAndCleanupUtterances(fileUtterances, filename)
     }
     return { convos: fileConvos, utterances: fileUtterances }
   }
 
+  _tagAndCleanupUtterances (utteranceFiles, filename) {
```
Comment (codeforequity-at) [inline src/scripting/ScriptingProvider.js]: that's a strange use of the "map" function - map applies a function on each element and returns the result. you should directly assign  the result of the map function the the uttterances local var.

Comment (stefan-sev) [reply src/scripting/ScriptingProvider.js]: I updated this. The reason why I return a new object is for avoiding side effects and being immutable. I updated to use Object.assign.

**Comment by codeforequity-at** [comment]
map creates new array implicit

Am So., 20. Jän. 2019, 12:37 hat Stefan Sevelda <notifications@github.com>
geschrieben:

File: src/scripting/ScriptingProvider.js (hunk 1 above)
Comment (stefan-sev) [reply src/scripting/ScriptingProvider.js]: You are right I just forgot the return

**Pull request merged**